### PR TITLE
Match `ftSeak_SpecialS`

### PR DESCRIPTION
**Pull request by DRGN-DRC**
~40% of the functions for Sheik have been disassembled (ftseak2.c complete). 
ftseak.c renamed to ftseak1.c, and the other files have been renamed appropriately (ftseak1 -> ftseak2, etc.).

### CODE REVIEW

**Comment by ribbanya** [comment]
@DRGN-DRC I'll have to clean up the git history before merging into `master`, but this should be a good starting point to get the functions linked.

**Review by ribbanya — CHANGES_REQUESTED** [review]
Unlinked functions need to be linked.

**Review by ribbanya — APPROVED** [review]
I got the dol to match by reverting `ftSeak_80110980`, `ftSeak_801110E8`, and `ftSeak_80111830`.

@DRGN-DRC up to you whether you want to get those to match before merging, otherwise we can just merge and start a new PR.

**Comment by DRGN-DRC** [comment]
I've got `ftSeak_80110980` to match. And almost `ftSeak_801110E8`. I'll try to see if I can finish these 3 up and then call it good to merge.

**Review by ribbanya — CHANGES_REQUESTED** [review]
This now depends on #745. I'd also like to finish cleaning up the `ftseak` files.

**Review by ribbanya — APPROVED** [review]
Okay this looks alright to me now. Still a lot of duplicated code but that can be dealt with later. Can you look it over, @r-burns?

File: src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c [hunk 1]
```diff
@@ -0,0 +1,1104 @@
...
+
+    {
+        Fighter* fp = GET_FIGHTER(fighter_gobj);
+
+        if (!fp->x2200_ftcmd_var0)
+            return;
+
+        {
+            u8* someptr = (u8*) fp + (12 * hitbox_id);
+            __memcpy(&someptr[0x2238], new_pos, 12);
```
Comment (r-burns) [inline src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c:216]: Going back and forth on whether we should fix these before merging. Should clearly be copying new_pos to a Vec3 in fp's special attrs indexed by hitbox_id, same for the memcpys below. Probably not a big deal if we merge and clean it up later.

File: src/melee/ft/types.h [hunk 2]
```diff
@@ -60,91 +60,122 @@ typedef enum FighterKind {
...
+    CKIND_FOX,       // Fox
+    CKIND_GAMEWATCH, // Mr. Game & Watch (GameWatch)
+    CKIND_KIRBY,     // Kirby
+    CKIND_KOOPA,     // Bowser (Koopa)
+    CKIND_LINK,      // Link
+    CKIND_LUIGI,     // Luigi
+    CKIND_MARIO,     // Mario
+    CKIND_MARS,      // Marth (Mars)
+    CKIND_MEWTWO,    // Mewtwo
     CKIND_NESS,      // Ness//
```
Comment (r-burns) [inline src/melee/ft/types.h:74]:
```suggestion
    CKIND_NESS,      // Ness
```
missed one

**Review by r-burns — COMMENTED** [review]
Looks pretty good, just a couple minor comments

File: src/melee/gr/grticeclimber.c [hunk 3]
```diff
@@ -157,8 +157,8 @@ void lbl_80221204(HSD_GObj* arg0) {}
 
 #ifdef MWERKS_GEKKO
 #pragma push
-asm void lbl_80221208(void)
-{ // clang-format off
+asm void lbl_80221208(void){
+    // clang-format off
```
[the remark this answers is not in the document]
Comment (ribbanya) [reply src/melee/gr/grticeclimber.c:161]: It's the one we've talked about before. Should I put the `format off` above the signature?

File: src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c (hunk 1 above)
Comment (ribbanya) [reply src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c:216]: Does `Fighter` already have all the necessary fields to do this? If so we should fix it now.

Comment (DRGN-DRC) [reply src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c:216]: I didn't like that either. I tried a few different syntax permutations to perform that copy, but ultimately that was the only one I could find that matched.

Comment (r-burns) [reply src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c:216]:
```suggestion
           fp->sa.seak.x2238[hitbox_id] = *new_pos;
```

**Review by r-burns — CHANGES_REQUESTED** [review]
Let's fix all these vector-indexed copies since we can do so easily using existing fighter fields. LGTM otherwise.

```diff
--- a/src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c
+++ b/src/melee/ft/chara/ftSeak/ftSeak_SpecialS.c
@@ -211,27 +211,18 @@ void ftSeak_UpdateHitboxes(HSD_GObj* fighter_gobj, Vec3* new_pos, s32 hitbox_id)
         if (!fp->x2200_ftcmd_var0)
             return;
 
-        {
-            u8* someptr = (u8*) fp + (12 * hitbox_id);
-            __memcpy(&someptr[0x2238], new_pos, 12);
+        fp->sa.seak.x2238[hitbox_id] = *new_pos;
 
-            if (new_pos->x != 0 || new_pos->y != 0) {
-                func_8007B8A8(&fp->x914[hitbox_id], new_pos);
+        if (new_pos->x != 0 || new_pos->y != 0) {
+            func_8007B8A8(&fp->x914[hitbox_id], new_pos);
 
-                {
-                    /// @todo Proper member offsets in Fighter
-                    f32* fltptr1 = (float*) ((s32) fp + (312 * hitbox_id));
-                    __memcpy(&fltptr1[600], new_pos, 12);
-
-                    {
-                        f32* fltptr2 = &fltptr1[603];
-                        if (fltptr1[603] == 0 && fltptr1[604] == 0 &&
-                            fltptr1[605] == 0)
-                        {
-                            __memcpy(fltptr2, new_pos, 12);
-                        }
-                    }
-                }
+            fp->x914[hitbox_id].x4C = *new_pos;
+
+            if (fp->x914[hitbox_id].x58.x == 0 &&
+                fp->x914[hitbox_id].x58.y == 0 &&
+                fp->x914[hitbox_id].x58.z == 0)
+            {
+                fp->x914[hitbox_id].x58 = *new_pos;
             }
         }
     }
```

File: src/melee/gr/grticeclimber.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (r-burns) [reply src/melee/gr/grticeclimber.c:161]: Ah right, let's not worry about it then.

**Review by ribbanya — DISMISSED** [review]
Force pushed to a clean history and fixed the vector copies.

**Review by ribbanya — APPROVED** [review]

**Review by r-burns — APPROVED** [review]

**Pull request merged**